sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetRowConverter.scala

```diff
@@ -202,8 +204,12 @@ private[parquet] class ParquetRowConverter(
 
   override def start(): Unit = {
     var i = 0
-    while (i < currentRow.numFields) {
+    while (i < fieldConverters.length) {
       fieldConverters(i).updater.start()
+      i += 1
+    }
+    i = 0
+    while (i < currentRow.numFields) {
```

Comment:
Can we loop once with if?

Reply:
Yes, but I think it's clearer this way. If @viirya has an opinion either way I'll take it as a "tie-breaker".